core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -80,8 +85,55 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
   //periodically emit metrics (don't need to do this for each message!)
   context.system.scheduler.schedule(30.seconds, 10.seconds, self, EmitMetrics)
 
+  // Key is ColdStartKey, value is the number of cold Start in minute
+  var coldStartCount = immutable.Map.empty[ColdStartKey, Int]
+
   backfillPrewarms(true)
 
+  // check periodically every 1 minute, adjust prewarmed container(delete if unused for some time and create some increment containers)
+  context.system.scheduler.schedule(1.minute, 1.minute, self, AdjustPrewarmedContainer)
+
+  def adjustPrewarmedContainer(): Unit = {
+    prewarmConfig.foreach { config =>
+      // Delete unused prewarmed container until minCount is reached
+      val kind = config.exec.kind
+      val memory = config.memoryLimit
+      val ttlSeconds = config.ttl.toSeconds
+      val minCount = config.minCount
+      val containers = prewarmedPool.filter { warmInfo =>
```

Comment:
suggest:
- change `PrewarmData.created:Instant` to `PrewarmData.expires: Option[Deadline]`
- add to PrewarmedData:`def isExpired(): Boolean = expires.exists(_.isOverdue())` 
- simplifying this removal logic to 
```
      prewarmedPool
        .filter { warmInfo =>
          warmInfo match {
            case (_, p @ PreWarmedData(_, `kind`, `memory`, _, _)) if p.isExpired() => true
            case _                                                                  => false
          }
        }
        .drop(config.minCount)
        .foreach(_._1 ! Remove)
```

Reply:
updated accordingly.